### PR TITLE
Fix cov shape handling

### LINKED ISSUE #857: Covariance data handled incorrectly in unSupervisedLearning
--------
Issue Description
--------
##### What did you expect to see happen?
That cov(x,y) == cov(y,x), but instead they are very different numbers.

##### What did you see instead?
cov(x,y) and cov(y,x), are very different numbers.

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.
Note that you have to add a print statement to see both covariance's:
framework/PostProcessors/TemporalDataMiningPostProcessor/Clustering.GaussianMixture
framework/PostProcessors/DataMiningPostProcessor/Clustering.GaussianMixture

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [ ] 1. Is it tagged with a type: defect or task?
- [ ] 2. Is it tagged with a priority: critical, normal or minor?
- [ ] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [ ] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [ ] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [ ] 1. If the issue is a defect, is the defect fixed?
- [ ] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [ ] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [ ] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [ ] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address?
Closes #857 

##### What are the significant changes in functionality due to this change request?
This changes the result of the datamining covariance when a 'full' covariance is requested.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [ ] 1. Review all computer code.
- [ ] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [ ] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [ ] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [ ] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [ ] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [ ] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by joshua-cogliati-inl** [comment]
Possibly the DataMining class should somehow get the type of the covariance to the unSupervised learning class, instead of looking at the shape.

**Comment by joshua-cogliati-inl** [comment]
@mandd Here is the merge request.

File: framework/unSupervisedLearning.py [hunk 1]
```diff
@@ -628,7 +628,12 @@ def __trainLocal__(self):
           rowSigma = self.muAndSigmaFeatures[rowFeature][1]
           for col, colFeature in enumerate(self.features):
             colSigma = self.muAndSigmaFeatures[colFeature][1]
-            covariance[row,col] = covariance[row,col] * rowSigma * colSigma
+            #if covariance type == full, the shape is (n_components, n_features, n_features)
+            if len(covariance.shape) == 3:
+              covariance[:,row,col] = covariance[:,row,col] * rowSigma * colSigma
+            else:
+              #XXX if covariance type == diag, this will be wrong.
```
[the remark this answers is not in the document]
Comment (joshua-cogliati-inl) [reply framework/unSupervisedLearning.py]: No, this comment is correct, this code will only work if covariance type  is either tied or full and will fail if type is diag

**Comment by alfoa** [comment]
Green light, but a proposed email should be attached...

**Comment by alfoa** [comment]
@joshua-cogliati-inl @mandd news?

**Comment by joshua-cogliati-inl** [comment]
No news from me.  (Well, it is fixed for type="full" but still broken if type="diag")

________________________________

**Comment by aalfonsi** [comment]
@joshua-cogliati-inl should we try to merge this?

**Comment by joshua-cogliati-inl** [comment]
> @joshua-cogliati-inl should we try to merge this?

Hm, we still need to figure out how to do "diag" (See XXX comment in changes.)

**Comment by joshua-cogliati-inl** [comment]
At some point, the change to fix covariance type = full got merged.  However, the other bug still exists.  Since this pull does not fix that, closing this pull, but leaving the issue open.

**Pull request closed without merging**